soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -6320,6 +6320,7 @@ s32 func_8083E5A8(Player* this, PlayState* play) {
                     Player_SetPendingFlag(this, play);
                     Message_StartTextbox(play, 0xF8, NULL);
                     Audio_PlayFanfare(NA_BGM_SMALL_ITEM_GET);
+                    GameInteractor_ExecuteOnReceiveItemHooks(ItemTable_RetrieveEntry(this->getItemEntry.modIndex, this->getItemEntry.getItemId));
```

Comment:
Same thing here, can we just pass in this->getItemEntry directly? And all the ones below in this file.